### PR TITLE
Enable Python 3.10

### PR DESCRIPTION
**Pull request by tonyabracadabra**
## Motivation and Context
Why was this PR created?

## How has this been tested?
What testing strategies have you used?

## Checklist

- [ ] Read the [contributing](https://github.com/quantumblacklabs/causalnex/blob/develop/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/causalnex/blob/develop/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Assigned myself to the PR

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by oentaryorj** [comment]
Thanks for your contribution @tonyabracadabra. Can we also make sure that the unit tests for 3.9 and 3.10 have the "required" tag (and that they run successfully)?

**Comment by oentaryorj** [comment]
Hi @tonyabracadabra, we can address the "required tag" later, but would you be able to fix the issues with `unit_tests_310` first please? Many thanks!

File: causalnex/utils/data_utils.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def count_unique_rows(data: pd.DataFrame, placeholder: float = -np.inf) -> pd.Da
     """
     # find a placeholder for NaNs: groupby excludes NaNs by default.
     # So we replace them with some value and later put the NaNs back
-    data.fillna(placeholder, inplace=True)
+    data = data.fillna(placeholder)
```
Comment (oentaryorj) [inline causalnex/utils/data_utils.py:113]: Any reason why we need to make a copy here?

File: tests/test_bayesiannetwork.py [hunk 2]
```diff
@@ -840,7 +840,7 @@ def test_bad_parent_node_states_index(self, bn, good_cpd):
         """Should raise an error when setting bad parent node states index"""
 
         bad_cpd = good_cpd
-        bad_cpd.columns.set_levels(["test1", "test2"], level=0, inplace=True)
+        bad_cpd.columns = bad_cpd.columns.set_levels(["test1", "test2"], level=0)
```
Comment (oentaryorj) [inline tests/test_bayesiannetwork.py:843]: Ditto ... any reason why we need to make a copy here?

**Comment by oentaryorj** [comment]
> > 
> 
> ok the test was not passed because of the coverage issue on line
> 
> ```python
> arg_spec = inspect.getargvalues(inspect.currentframe())
> ```
> 
> I am not sure why it is not covered in testing, seems that `getargvalues` has been marked deprecated a while ago, I added a `# pragma: no cover` to make the coverage 100%, let me know if this is acceptable or not.

Yes, this is fine. If you can suggest an alternative to this deprecated function, though, it would be great! :)

File: causalnex/structure/pytorch/sklearn/_base.py [hunk 3]
```diff
@@ -205,7 +205,7 @@ def fit(self, X: Union[pd.DataFrame, np.ndarray], y: Union[pd.Series, np.ndarray
             X = X.copy()
             self._ss_X = StandardScaler()
             X.iloc[:, self.continuous_col_idxs] = self._ss_X.fit_transform(
-                X.iloc[:, self.continuous_col_idxs]
+                X.iloc[:, self.continuous_col_idxs].values
```
Comment (oentaryorj) [inline causalnex/structure/pytorch/sklearn/_base.py:208]: Sorry, why is this change necessary?

Comment (oentaryorj) [reply causalnex/structure/pytorch/sklearn/_base.py:208]: Got it thanks!

File: tests/test_bayesiannetwork.py (hunk 2 above)
Comment (oentaryorj) [reply tests/test_bayesiannetwork.py:843]: Ok noted. Thank you!

File: causalnex/utils/data_utils.py (hunk 1 above)
Comment (oentaryorj) [reply causalnex/utils/data_utils.py:113]: OK noted. thanks!

**Review by oentaryorj — APPROVED** [review]

**Comment by oentaryorj** [comment]
> Will we have a release soon for the recent support on python 3.9, 3.10? @oentaryorj

We shall publish a new release once we have enough new features and/or bug fixes.

**Review by EvgeniiaVeselova — APPROVED** [review]
LGTM
